packages/querygrid/src/components/user/actions.ts

```diff
@@ -0,0 +1,107 @@
+import moment from 'moment'
+import { Map, OrderedMap } from 'immutable'
+import { Ajax , ActionURL, Utils } from '@labkey/api'
+import { caseInsensitive, hasAllPermissions, PermissionTypes, SchemaQuery, User, buildURL } from "@glass/base";
+import { ChangePasswordModel } from "./models";
+
+export function getUserPermissionsDisplay(user: User): Array<string> {
+    let permissions = [];
+
+    if (user.isAdmin) {
+        permissions.push('Administrator');
+    }
+    else {
+        if (hasAllPermissions(user, [PermissionTypes.DesignSampleSet])) {
+            permissions.push('Sample Set Designer');
+        }
+        if (hasAllPermissions(user, [PermissionTypes.DesignAssay])) {
+            permissions.push('Assay Designer');
+        }
+        permissions.push(user.canUpdate ? 'Editor' : (user.canInsert ? 'Author' : 'Reader'));
```

Comment:
I think "Author" should be removed here.

Reply:
Yes, I was going to wait for the SM user permissions story to touch that part.